### PR TITLE
Use classes for image type identification code

### PR DESCRIPTION
**Pull request by zas**
# Summary

* This is a…
    * [ ] Bug fix
    * [ ] Feature addition
    * [x] Refactoring
    * [ ] Minor / simple change (like a typo)
    * [ ] Other
* **Describe this change in 1-2 sentences**:

# Problem

* JIRA ticket (_optional_): PICARD-XXX

# Solution

# Action

### CODE REVIEW

File: picard/util/imageinfo.py [hunk 1]
```diff
@@ -43,189 +43,236 @@ class UnexpectedError(IdentificationError):
...
+        def _tiff_read_value(type, order, data):
+            if type == TIFF_TYPE_LONG:
+                value = data[:4]
+                format = order + 'I'
+            elif type == TIFF_TYPE_SHORT:
+                value = data[:2]
+                format = order + 'H'
+            else:
+                raise UnexpectedError('TIFF: unexpected field type %s' % type)
+            return struct.unpack(format, value)[0]
```
Comment (phw) [inline picard/util/imageinfo.py:221]: I would not define this inside the function but on the class itself (`self._read_value`, the tiff prefix can also be dropped).

Same probably for the constants.

**Review by phw — CHANGES_REQUESTED** [review]
Well done. I would restructure the TIFF class a bit to not define everything inside the read function, but otherwise LGTM

File: picard/util/imageinfo.py (hunk 1 above)
Comment (zas) [reply picard/util/imageinfo.py]: Agreed, done in 7bf8e06

**Review by phw — APPROVED** [review]
Cool. Much clearer to extend for new formats in the future.

**Pull request merged**